drivers/watchdog/wdt_counter.c

```diff
@@ -56,6 +57,7 @@ static void counter_alarm_callback(const struct device *dev,
 	struct wdt_counter_data *data = wdt_dev->data;
 
 	counter_stop(dev);
+	data->used[chan_id] = false;
```

Comment:
The assumption that a channel can be freed as soon as the callback was triggered is not documented anywhere. Is it the same across other watchdog drivers? I don't think so.

Reply:
You are right. Most drivers don't support watchdog channels. NRFX watchdog driver doesn't reuse channels.

Would you accept patch which makes SOC reset as an option https://github.com/zephyrproject-rtos/zephyr/pull/53195/commits/8b70d91bef88284462e2be69efbfc5463dc33bbd. If yes, I will remove free channels tracking from this PR.

Reply:
Yes, that sounds good. Let's keep the channel tracking out for now.

Reply:
Done! Please check if it's good now.